src/App.tsx

```diff
@@ -1,61 +1,147 @@
+import React, { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { User } from './types/User';
+import { Todo } from './types';
+
+const getUserById = (userId: number) => {
+  return usersFromServer.find(user => user.id === userId)
+  || usersFromServer;
+};
+
+const preparedTodos = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId) as User,
+}));
 
 export const App = () => {
+  const [todos, setTodos] = useState<Todo[]>(preparedTodos);
+  const [hasError, setHasError] = useState({ title: false, userId: false });
+  const [values, setValues] = useState({
+    title: '',
+    userId: 0,
+  });
+
+  const { title, userId } = values;
+
+  const createNewId = () => {
+    const allId = todos.map(todo => todo.id);
+
+    return Math.max(...allId) + 1;
+  };
+
+  const handleTitleInput = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setValues({ ...values, title: e.target.value });
+
+    setHasError({ ...hasError, title: false });
+  };
+
+  const handleUserIdSelect = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    setValues({ ...values, userId: +e.target.value });
+
+    setHasError({ ...hasError, userId: false });
+  };
+
+  const reset = () => {
+    setHasError({ title: false, userId: false });
+
+    setValues({
+      title: '',
+      userId: 0,
+    });
+  };
+
+  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    if (title === '' && userId === 0) {
+      setHasError({ title: true, userId: true });
+
+      return;
+    }
+
+    if (title === '') {
+      setHasError({ ...hasError, title: true });
+
+      return;
+    }
+
+    if (userId === 0) {
+      setHasError({ ...hasError, userId: true });
+
+      return;
+    }
+
+    const newTodo = {
+      id: createNewId(),
+      title,
+      userId,
+      completed: false,
+      user: getUserById(userId) as User,
```

Comment:
```suggestion
      user: getUserById(userId),
```

better to avoid assertions as much as you can